File: 1018.py
def start_point(A, B):
    count1, count2 = 0, 0
    for y in range(A, A + 8):
        for x in range(B, B + 8):
            if (x + y) % 2 == 0:
                if matrix[x][y] == 'B':
                    count1 += 1
            else:
                if matrix[x][y] == 'W':
                    count1 += 1

    for y in range(A, A + 8):
        for x in range(B, B + 8):
            if (x + y) % 2 == 0:
                if matrix[x][y] == 'W':
                    count2 += 1
            else:
                if matrix[x][y] == 'B':
                    count2 += 1
    min_list.append(count1)
    min_list.append(count2)

min_list = []
R, C = map(int, input().split())
# 세로만큼의 입력을 추가로 한줄씩 받아온다
matrix = [[0] * C for i in range(R)]
for j in range(0, int(R)):
    tempstring = input()
    for i in range(0, int(C)):
        # 받아온데이터를 map배열에 입력
        matrix[j][i] = tempstring[i]

for j in range(0, R - 7):
    for i in range(0, C - 7):
        start_point(i, j)

print(min(min_list))

File: 1018.js
const fs = require('fs');
const input = fs.readFileSync("/dev/stdin").toString().trim().split("\n");
// let input = "8 8\nWBWBWBWB\nBWBWBWBW\nWBWBWBWB\nBWBBBWBW\nWBWBWBWB\nBWBWBWBW\nWBWBWBWB\nBWBWBWBW";
// input = input.split("\n");
const firstLine = input[0];
const N = parseInt(firstLine.split(" ")[0]);
const M = parseInt(firstLine.split(" ")[1]);

let n_arr;
for (let i = 0; i < N; i += 1) {
    n_arr = input.slice(1, N + 1);
}

function cal(x, y) {
    let result1 = 0;
    let result2 = 0;
    for (let i = x; i < x + 8; i += 1) {
        for (let j = y; j < y + 8; j += 1) {
            if ((i + j) % 2 == 0) {
                if (n_arr[j][i] == 'B') {
                    result1 += 1;
                }
            }
            else {
                if (n_arr[j][i] == 'W') {
                    result1 += 1;
                }
            }
        }
    }

    for (let i = x; i < x + 8; i += 1) {
        for (let j = y; j < y + 8; j += 1) {
            if ((i + j) % 2 == 0) {
                if (n_arr[j][i] == 'W') {
                    result2 += 1;
                }
            }
            else {
                if (n_arr[j][i] == 'B') {
                    result2 += 1;
                }
            }
        }
    }
    return Math.min(result1, result2);
}


let result = [];
for (let i = 0; i < N - 7; i += 1) {
    for (let j = 0; i < M - 7; i += 1) {
        result.push(cal(i, j));
    }
}
console.dir(Math.min(result));
// const n_arr = input.slice(1, n + 1);
// const [m, ...m_arr] = input.slice(n + 1);


